src/lib.rs

```diff
@@ -1491,10 +1491,7 @@ impl Build {
                     Some(true) => "-MT",
                     Some(false) => "-MD",
                     None => {
-                        let features = self
-                            .getenv("CARGO_CFG_TARGET_FEATURE")
-                            .unwrap_or(String::new());
-                        if features.contains("crt-static") {
+                        if cfg!(target_feature = "crt-static") {
```

Comment:
There is a reason for why it's inappropriate to use `cfg!` here. Keyword is that it applies to the build script itself, not the compilation target.

Reply:
My tests with these changes were able to properly infer the C runtime based on the settings in my config.toml from the parent crate. 

The cc build script gets compiled during the build invocation of the parent crate which means the cgf! directive has access to the parent crate target_feature no ?

Reply:
The build script is compiled for the platform that executes the Rust compiler, but it doesn't necessarily mean that the compiler was asked to generate code for the same platform. Well, one can argue that `*-windows-msvc` binary code is generated only by `*-windows-msvc` toolchain, but formally speaking it's not safe assumption to make. And cc-rs doesn't make it.